crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -865,6 +865,10 @@ where
                     // Special sync mode where verified blocks were inserted into blockchain
                     // directly, archiving of this block will naturally happen later
                     continue;
+                } else if best_archived_block_number.is_zero() {
+                    // We may have imported some block using special sync mode right after genesis,
+                    // in which case archiver will be stuck at genesis block
+                    continue;
```

Comment:
@shamil-gadelshin I was debugging code in my head instead of sleeping yesterday and identified that this is a special case for genesis only and will not work in case we implement snap sync as a generic sync mechanism (which I hope is the next step?).

We'll need to change it, for example to check if parent block is even present at all instead, which will both be the case for genesis and snap sync started after some history is already present.

Reply:
We can discuss that later when we start that task. I might even convince you to get back to explicit component notifications in contrast with implicit built-in special cases.